app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-ecommerce-tracker.js

```diff
@@ -9,12 +9,17 @@
     PIIRemover: new GOVUK.analyticsGa4.PIIRemover(),
     DEFAULT_LIST_TITLE: 'Site search results',
 
-    init: function (isNewPageLoad) {
+    init: function (dynamicPageViewData) {
       if (window.dataLayer) {
+        /* The dynamicPageViewData object is only passed to the init() function as a result of an AJAX request
+        (in live_search.js in the finder-frontend repository). Otherwise it will not be available and this indicates a fresh page load.
+        This is needed to trigger a fresh pageViewTracker push to the dataLayer on dynamic page updates (line 30) and to prevent multiple
```

Comment:
This is a helpful comment. My only thought is that having specific line number references may get out of date very quickly, could these instead refer to functions? Might last longer.

Reply:
That’s very true, I’ve updated the comment